aws/resource_aws_lambda_function.go

```diff
@@ -139,6 +139,31 @@ func resourceAwsLambdaFunction() *schema.Resource {
 						},
 					},
 				},
+
+				// Suppress diffs if the VPC configuration is empty.
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if v, ok := d.GetOk("vpc_config"); ok {
+						configs := v.([]interface{})
+						config, ok := configs[0].(map[string]interface{})
```

Comment:
Thanks @joshuaspence for submitting this with an acceptance test! This was definitely on the right track.

One note here: to prevent potential panics, we should ensure the length of `configs` before trying to reference the first element, e.g.

```go
if v, ok := d.GetOk("vpc_config"); ok && len(v.([]interface{})) > 0 {
```

Style nitpick: Also, we can remove the nesting if return early:

```
if old != "0" && new != "1" {
  return false
}
v, ok := d.GetOk("vpc_config")
if !ok || len(v.([]interface{})) == 0 {
  return false
}
// other logic :)
```

I'll be rebasing this PR and applying a followup commit so we can get this released today, rather then making you do any additional work or having this linger further. 👍